ptn/boozebot/botcommands/Departures.py

```diff
@@ -0,0 +1,292 @@
+            f'location: {departure_location} to {arrival_location}.')
+        
+        # Defer the interaction response to allow more time for processing
+        await interaction.response.defer()
+
+        # Convert carrier ID to uppercase
+        carrier_id = carrier_id.upper()
+
+        # Validate the carrier ID format
+        if not re.match(r"\w{3}-\w{3}", carrier_id):
+            print(f'{interaction.user.name}, the carrier ID was invalid, XXX-XXX expected received, {carrier_id}.')
+            return await interaction.edit_original_response(content=f'{interaction.user.name}, the carrier ID was invalid during tanker unload, '
+                                        f'XXX-XXX expected received, {carrier_id}.')
+
+        # Acquire the database lock and fetch carrier data
+        pirate_steve_lock.acquire()
+        pirate_steve_db.execute(
+            "SELECT * FROM boozecarriers WHERE carrierid LIKE (?)", (f'%{carrier_id}%',)
+        )
+
+        carrier_data = pirate_steve_db.fetchone()
+        pirate_steve_lock.release()
+        
+        # Check if carrier data was found
+        if not carrier_data:
+            print(f'We failed to find the carrier: {carrier_id} in the database.')
+            return await interaction.edit_original_response(content=f'Sorry, during unload we could not find a carrier for the data: {carrier_id}.')
+
+        # Create a BoozeCarrier object from the fetched data
+        carrier_data = BoozeCarrier(carrier_data)
+        
+        carrier_name = carrier_data.carrier_name
+        carrier_id = carrier_data.carrier_identifier
+        
+        # Function to sanitize input by removing certain characters
+        def sanitize_input(input):
+            return input.replace("<", "").replace(">", "").replace("@", "").replace("|", "")
+        
+        # Sanitize the input data
+        carrier_name = sanitize_input(carrier_name)
+        carrier_id = sanitize_input(carrier_id)
+        arrival_location = sanitize_input(arrival_location)
+        departure_location = sanitize_input(departure_location)
+        
+        departure_time_text = ""
+        
+        # Handle departure time if provided as a timestamp
+        if departing_at:
+            try:
+                departure_timestamp = departing_at
+                if departure_timestamp.startswith("<t:") and departure_timestamp.endswith(">"):
+                    departure_timestamp = departure_timestamp.split(":")[1]
+                departure_timestamp = int(departure_timestamp)
+            except ValueError:
+                print(f"Departure time was not an integer: {departing_at}")
+                return await interaction.edit_original_response(content=f"Departure time was not a valid timestamp: {departing_at}. You can use <https://hammertime.cyou> to generate them.")
+            
+            # Validate the timestamp range
+            if not (departure_timestamp > 0 and departure_timestamp < 2147483647):
```

Comment:
suggestion(non-blocking): limit even further. Min is now, max is now+X (1 week?)